tests/jvm/src/test/scala/munit/TestSelectorSuite.scala

```diff
@@ -0,0 +1,122 @@
+package munit
+
+import sbt.testing.Event
+import sbt.testing.EventHandler
+import sbt.testing.Selector
+import sbt.testing.Status
+import sbt.testing.SuiteSelector
+import sbt.testing.TaskDef
+import sbt.testing.TestSelector
+import scala.collection.mutable
+import sbt.testing.Runner
+
+/**
+ * Dummy test suite which is needed by TestSelectorSuite
+ */
+class MyTestSuite extends FunSuite {
+
+  test("testFoo") {
+    assert(true, "Test should pass")
+  }
+
+  test("testBar") {
+    assert(true, "Test should pass")
+  }
+}
+
+/**
+ * Check if TestSelector's are correctly handled by Munit.
+ * Execute prepared TaskDef's using manually created instances of sbt.testing.{Framework and Runner}.
+ */
+class TestSelectorSuite extends FunSuite {
+  val framework = new Framework();
+  val runner: Runner = framework.runner(
+    Array.empty,
+    Array.empty,
+    this.getClass().getClassLoader()
+  );
+
+  val fingerprint = framework.munitFingerprint
```

Comment:
So this test mimics the behavior of sbt's `ForkMain` - it creates a framework instance which later will be used for running tests.